Kernel/ACPI/Parser.h

```diff
@@ -90,6 +97,8 @@ class Parser {
     Vector<PhysicalAddress> m_sdt_pointers;
     PhysicalAddress m_fadt;
     PhysicalAddress m_facs;
+    u64 m_dsdt { 0 };
```

Comment:
Is `m_dsdt` supposed to indicate the physical address of the the DSDT table? if so, why not to use `PhysicalAddress` instead?
Also, remember that the FADT holds this value twice (one for ACPI 1, and an extended pointer for later revisions).

Reply:
Yeah eventually. The problem is that you can have 64 bit physical addresses on a 32 bit machine, but we use `FlatPtr` only. In fact, there's two DSDT pointers, an optional 64 bit physical address and the legacy 32 bit physical address.

Reply:
Then I think we should make `PhysicalAddress` to wrap a `u64` value as it represents a physical address which can be as you said 64 bits on 32 bit machine.
About the DSDT pointer, AFAIK most machines use both DSDT pointers to point to the same physical address as it doesn't make sense to make two copies of the same data. On my machine I decompiled the FADT blob and saw that this is true.